lib/nerves_network/if_supervisor.ex

```diff
@@ -35,31 +35,95 @@ defmodule Nerves.Network.IFSupervisor do
 
   def setup(ifname, settings) do
     manager_modules = managers(if_type(ifname), settings)
-    Logger.debug fn -> "#{__MODULE__} manager_modules: #{inspect manager_modules}" end
+    Logger.debug(".setup manager_modules: #{inspect manager_modules}")
 
     children =
       for manager <- manager_modules  do
         child_name = pname(ifname, manager)
         worker(manager, [ifname, settings, [name: child_name]], [id: {pname(ifname), child_name}, restart: restart_type(manager)])
       end
-    Logger.debug fn -> "#{__MODULE__} children: #{inspect children}" end
+
+    Logger.debug(".setup children: #{inspect children}")
+
+    result =
+      {:ok, for child <- children do
+              Logger.debug(".setup Starting child: #{inspect child}...")
+              retval = Supervisor.start_child(__MODULE__, child)
+              Logger.debug(".setup retval = #{inspect retval}...")
+              retval
+            end #For child <- children
+      }
+
+    Logger.debug(".setup returns #{inspect result}")
+  end
+
+
+
+  @doc """
+  Returns `{:ok, list()}`.
+
+  ## Parameters
+  - ifname: String identifying network interface's name i.e. "eth0"
+  - settings: a Keyword list with the settings i.e. [ipv4_address_method: :dhcp, ipv6_dhcp: stateless]. For stop function the key values are
+            irelevant because the settings are used for only locating an appropriate manager tied to the network interface specified with ifname.
+
+  ## Examples
+
+        iex> Nerves.Network.IFSupervisor.stop("eth0", [ipv6_dhcp: :stateless])
+        {:ok, [ok: :ok]}
```

Comment:
Same here--I think the return should be `:ok` or communicate more what the list is that is being returned

Reply:
you see the stop comprises of two function's calls... we do not panic when they fail i.e. {:error, :not_found} so we can basicall blindly call stop() then setup() without the necessity of process being stopped to really be there.